### PR TITLE
OSDOCS-3870: Ported the Ingress content to OSD/ROSA

### PR DESCRIPTION
**Pull request by EricPonvelle**
Version(s):
Enterprise 4.10+

Issue:
https://issues.redhat.com/browse/OSDOCS-3870

Link to docs preview:

- OCP is unchanged
    - [Ingress Operator in OpenShift Container Platform](http://file.rdu.redhat.com/eponvell/OSDOCS-3870_IngressPort/enterprise/networking/ingress-operator.html)
    - [Configuring the Ingress Controller endpoint publishing strategy](http://file.rdu.redhat.com/eponvell/OSDOCS-3870_IngressPort/enterprise/networking/nw-ingress-controller-endpoint-publishing-strategies.html)
    - Configuring ingress cluster traffic:
        - [Overview](http://file.rdu.redhat.com/eponvell/OSDOCS-3870_IngressPort/enterprise/networking/configuring_ingress_cluster_traffic/overview-traffic.html)
        - [Configuring ExternalIPs for services](http://file.rdu.redhat.com/eponvell/OSDOCS-3870_IngressPort/enterprise/networking/configuring_ingress_cluster_traffic/configuring-externalip.html)
        - [Configuring ingress cluster traffic using an Ingress Controller](http://file.rdu.redhat.com/eponvell/OSDOCS-3870_IngressPort/enterprise/networking/configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-ingress-controller.html)
        - [Configuring ingress cluster traffic using a load balancer](http://file.rdu.redhat.com/eponvell/OSDOCS-3870_IngressPort/enterprise/networking/configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-load-balancer.html)
        - [Configuring ingress cluster traffic on AWS](http://file.rdu.redhat.com/eponvell/OSDOCS-3870_IngressPort/enterprise/networking/configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-aws.html)
        - [Configuring ingress cluster traffic using a service external IP](http://file.rdu.redhat.com/eponvell/OSDOCS-3870_IngressPort/enterprise/networking/configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-service-external-ip.html)
        - [Configuring ingress cluster traffic using a NodePort](http://file.rdu.redhat.com/eponvell/OSDOCS-3870_IngressPort/enterprise/networking/configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-nodeport.html)
- ROSA:
    - [Ingress Operator in Red Hat OpenShift Service on AWS](http://file.rdu.redhat.com/eponvell/OSDOCS-3870_IngressPort/rosa/networking/ingress-operator.html)
    - ~~Configuring the Ingress Controller endpoint publishing~~
    - ~~Configuring ingress cluster traffic~~:
        - ~~Overview~~
        - ~~Configuring ExternalIPs for services~~
        - ~~Configuring ingress cluster traffic using an Ingress Controller~~
        - ~~Configuring ingress cluster traffic using a load balancer~~
        - ~~Configuring ingress cluster traffic on AWS~~
        - ~~Configuring ingress cluster traffic using a service external IP~~
        - ~~Configuring ingress cluster traffic using a NodePort~~
- OSD:
    - [Ingress Operator in OpenShift Dedicated](http://file.rdu.redhat.com/eponvell/OSDOCS-3870_IngressPort/dedicated/networking/ingress-operator.html)
    - ~~Configuring the Ingress Controller endpoint publishing strategy~~
    - ~~Configuring ingress cluster traffic~~:
        - ~~Overview~~
        - ~~Configuring ExternalIPs for services~~
        - ~~Configuring ingress cluster traffic using an Ingress Controller~~
        - ~~Configuring ingress cluster traffic using a load balancer~~
        - ~~Configuring ingress cluster traffic on AWS~~
        - ~~Configuring ingress cluster traffic using a service external IP~~
        - ~~Configuring ingress cluster traffic using a NodePort~~
    

Additional information:
Ported the Ingress information from OCP to OSD/ROSA. This information hasn't been curated.

### CODE REVIEW

**Comment by bergerhoffer** [comment]
The enterprise-4.12 label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.11. And any PR going into `main` must also target the latest version branch (enterprise-4.12).

If the update in your PR does NOT apply to version 4.12 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

File: _topic_maps/_topic_map_osd.yml [hunk 1]
```diff
@@ -186,11 +186,32 @@ Name: Networking
 Dir: networking
 Distros: openshift-dedicated
 Topics:
+- Name: Understanding the Ingress Operator
+  File: ingress-operator
+- Name: Configuring the Ingress Controller endpoint publishing strategy
```
Comment (dustman9000) [inline _topic_maps/_topic_map_osd.yml:191]: This should not be configurable by the customer. We should state that this is SRE managed via OCM+Hive.

File: _topic_maps/_topic_map_osd.yml [hunk 2]
```diff
@@ -186,11 +186,32 @@ Name: Networking
...
+  File: nw-ingress-controller-endpoint-publishing-strategies
 - Name: OpenShift SDN default CNI network provider
   Dir: openshift_sdn
   Topics:
   - Name: Enabling multicast for a project
     File: enabling-multicast
+- Name: Configuring ingress cluster traffic
+  Dir: configuring_ingress_cluster_traffic
+  Topics:
+  - Name: Overview
```
Comment (dustman9000) [inline _topic_maps/_topic_map_osd.yml:201]: This overview page is describing features like MetalLB and IP failover that is not relevant to OSD/ROSA. I think we should just omit this whole "Configuring ingress cluster traffic" section to not confuse the customer.

File: _topic_maps/_topic_map_osd.yml [hunk 3]
```diff
@@ -186,11 +186,32 @@ Name: Networking
 Dir: networking
 Distros: openshift-dedicated
 Topics:
+- Name: Understanding the Ingress Operator
```
Comment (dustman9000) [inline _topic_maps/_topic_map_osd.yml:189]: I think this is the only section that provides value and doesn't confuse the customer. I think the rest of the sections should be omitted as those are either not relevant or are SRE-managed resources.

**Comment by dustman9000** [comment]
Hi @EricPonvelle - I'm all done reviewing. I think the only valuable section to keep is the "Understanding the Ingress Operator" section. All other sections are either not relevant or are resources that are SRE-managed.

**Comment by dustman9000** [comment]
@EricPonvelle - For the TOC and OCP content, please only show the following:
image
and
image

**Comment by dustman9000** [comment]
TOC looks good, thanks!
/lgtm
/approve

**Review by opayne1 — COMMENTED** [review]
/lgtm

**Pull request merged**